website/notifications/events/model.py

```diff
@@ -0,0 +1,352 @@
+    """
+    def __init__(self, user, node, event, payload=None):
+        super(ComplexFileEvent, self).__init__(user, node, event, payload=payload)
+
+        self.source_node = Node.load(self.payload['source']['node']['_id'])
+        self.addon = self.node.get_addon(self.payload['destination']['provider'])
+
+    def _build_message(self, html=False):
+        addon, f_type, action = tuple(self.action.split("_"))
+        # f_type is always file for the action
+        if self.payload['destination']['kind'] == u'folder':
+            f_type = 'folder'
+
+        destination_name = self.payload['destination']['materialized'].lstrip('/')
+        source_name = self.payload['source']['materialized'].lstrip('/')
+
+        if html:
+            return (
+                '{action} {f_type} "<b>{source_name}</b>" '
+                'from {source_addon} in {source_node_title} '
+                'to "<b>{dest_name}</b>" in {dest_addon} in {dest_node_title}.'
+            ).format(
+                action=action,
+                f_type=f_type,
+                source_name=source_name,
+                source_addon=self.payload['source']['addon'],
+                source_node_title=self.payload['source']['node']['title'],
+                dest_name=destination_name,
+                dest_addon=self.payload['destination']['addon'],
+                dest_node_title=self.payload['destination']['node']['title'],
+            )
+        return (
+            '{action} {f_type} "{source_name}" '
+            'from {source_addon} in {source_node_title} '
+            'to "{dest_name}" in {dest_addon} in {dest_node_title}.'
+        ).format(
+            action=action,
+            f_type=f_type,
+            source_name=source_name,
+            source_addon=self.payload['source']['addon'],
+            source_node_title=self.payload['source']['node']['title'],
+            dest_name=destination_name,
+            dest_addon=self.payload['destination']['addon'],
+            dest_node_title=self.payload['destination']['node']['title'],
+        )
+
+    @property
+    def html_message(self):
+        return self._build_message(html=True)
+
+    @property
+    def text_message(self):
+        return self._build_message(html=False)
+
+    @property
+    def waterbutler_id(self):
+        return self.payload['destination']['path'].strip('/')
+
+    @property
+    def event(self):
+        """Sets event to be passed as well as the source event."""
+        if self.payload['destination']['kind'] != u'folder':
+            return '{}_file_updated'.format(self.waterbutler_id)  # file
+
+        return 'file_updated'  # folder
+
+    @property
+    def source_url(self):
+        url = furl(self.source_node.absolute_url)
+        url.path.segments = self.source_node.web_url_for('collect_file_trees').split('/')
+
+        return url
+
+
+class AddonFileMoved(ComplexFileEvent):
```

Comment:
Is it always 80 or is it 120? I guess this depends on the history of punchcards we ascribe to.